strongback/src/org/strongback/AsyncSwitchReactor.java

```diff
@@ -44,22 +38,30 @@ public void execute(long time) {
 
     @Override
     public void onTriggered(Switch swtch, Runnable function) {
-        listeners.getOrDefault(swtch, new Container()).addWhenTriggered(function);
+        Container listener = listeners.getOrDefault(swtch, new Container());
+        listener.addWhenTriggered(function);
+        listeners.putIfAbsent(swtch, listener);
```

Comment:
Very good. We should have used `Map.compute(...)` rather than `Map.getOrDefault(...)`, since the former actually does update the entry and the latter does not.